### PR TITLE
Worker pool cleanups

### PR DESCRIPTION
**Pull request by NelsonVides**
This PR is a general cleanup. It removes an unused header file, removes long-begone comments, recovers a configuration key for the pool sizes, increases symmetry between both async workers, and perhaps the only functional change, it stores the pool name on the same record that stores the pool details, instead of continuously generating the name again and again – this is a change in line with https://github.com/inaka/worker_pool/pull/181.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/90694) / small_tests / 928eb74c88864e74acad7f72175495dbb911392c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3419/90694/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3419/90694/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-11-29_10.13.02/lib.mongooseim.logs/run.2021-11-29_10.13.03/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/90708) / small_tests / 80b1197b084beae416091cb52e3c3fdf0403ebe5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3419/90708/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3419/90708/small_tests.23.1/small/ct_run.mongooseim@localhost.2021-11-29_10.32.18/lib.mongooseim.logs/run.2021-11-29_10.32.19/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/90737) / small_tests / 301a30d5b2f0ad10fa4c770fabeddbc46388a2c9
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3419/90737/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3419/90737/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-11-29_11.14.07/lib.mongooseim.logs/run.2021-11-29_11.14.08/suite.log.html)

File: src/mongoose_hooks.erl [hunk 1]
```diff
@@ -1008,6 +1009,14 @@ mam_lookup_messages(HookServer, Params) ->
 mam_archive_message(HookServer, Params) ->
     run_hook_for_host_type(mam_archive_message, HookServer, ok, [HookServer, Params]).
 
+%%% @doc The `mam_flush_messages' hook is run after the async bulk write
+%%% happens for messages despite the result of the write.
+-spec mam_flush_messages(HookServer :: jid:lserver(),
+                             MessageCount :: integer()) -> ok.
```
Comment (chrzaszcz) [inline src/mongoose_hooks.erl:1015]:
```suggestion
                         MessageCount :: integer()) -> ok.
```

**Review by chrzaszcz — COMMENTED** [review]
Looks good. I just think it would be good to use the config defaults when introducing new options (and add a test line to test the default in `config_parser_SUITE`).

**Comment by NelsonVides** [comment]
> Looks good. I just think it would be good to use the config defaults when introducing new options (and add a test line to test the default in `config_parser_SUITE`).

Sounds good, but, how to do so exactly? I thought we aren't doing defaults for modules yet, so far only global options were done, right? 🤔

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/90759) / small_tests / eee8837c2ec5f1844b5f71b913687deff37d7943
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3419/90759/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3419/90759/small_tests.23.1/small/ct_run.mongooseim@localhost.2021-11-29_13.20.33/lib.mongooseim.logs/run.2021-11-29_13.20.34/suite.log.html)

**Comment by chrzaszcz** [comment]
> Sounds good, but, how to do so exactly? I thought we aren't doing defaults for modules yet, so far only global options were done, right? 🤔
Let's leave it for later as I've looked at the `muc` and `pm` sections and they make it non-trivial (defaults would make the sections always enabled...)

**Review by arcusfelis — APPROVED** [review]
ok

**Comment by chrzaszcz** [comment]
Actually I think (correct me if I'm wrong) that this pool size option does nothing, the pool size is always 32 regardless of what I set there (in pm, muc or globally), because this option is never propagated to the async writer module (as some other options that are already there).

**Comment by NelsonVides** [comment]
As discussed with @chrzaszcz, turns out many options are not passed to the async workers, but this was broken before this PR, so here no bugs are introduced, and correctly passing the params to the workers can be done in a separate PR as a bug-fix.

**Pull request merged**